frontend/src/modules/SimulationTimeSeriesMatrix/utils/ensembleDisplayName.ts

```diff
@@ -0,0 +1,26 @@
+import { EnsembleIdent } from "@framework/EnsembleIdent";
+import { EnsembleSet } from "@framework/EnsembleSet";
+
+export class EnsembleDisplayNameGenerator {
+    private _ensembleSet: EnsembleSet;
+
+    constructor(ensembleSet: EnsembleSet) {
+        this._ensembleSet = ensembleSet;
+    }
+
+    getEnsembleDisplayName(ensembleIdent: EnsembleIdent): string {
+        const ensembleNameCount = this._ensembleSet
+            .getEnsembleArr()
+            .filter((ensemble) => ensemble.getEnsembleName() === ensembleIdent.getEnsembleName()).length;
+        if (ensembleNameCount === 1) {
+            return ensembleIdent.getEnsembleName();
+        }
+
+        const ensemble = this._ensembleSet.findEnsemble(ensembleIdent);
+        if (!ensemble) {
+            return ensembleIdent.getEnsembleName();
+        }
+
+        return ensemble.getDisplayName();
+    }
+}
```

Comment:
I see the point of separating for responsibility and that is a good approach. However, here we are a bit in the middle between advantages by division by responsibility, boilerplate/complexity, and over-generalizing (for now). Thinking more about it, what would stop us from moving the function inside the `EnsembleSet` class directly? IMO, that would be the cleanest and most comprehensive approach, also when it comes to responsibility. What do you think?

Reply:
Not very important to me, but now I moved the implementation of `EnsembleDisplayNameGenerator` into the `subplotBuilder`. My goal is to prevent passing `Ensemble` objects or `EnsembleSet` directly to the builder, as I prefer to work with a list of `VectorSpec` objects - which contain `EnsembleIdent` - and have a "lookup" of display name for the `EnsembleIdent`.

We can of course move it to the `EnsembleSet` class, but I feel I'm the only needing this as of now, so I don't feel for putting it there before it's a need for more modules. 

Note: If the code is moved to the `EnsembleSet` class I'm still in need of providing an `EnsembleSet` object containing only the selected `ensembles` to obtain wanted functionality - i.e. excluding `case_name` if there is no duplicated ensemble names. And I have to provide some callback function (or other solution for "lookup") to pass into the `subplotBuilder`.

Reply:
To me a class can be an easy way to abstract and wrap functionality - as here. Rather than passing an `EnsembleSet` to another class, in my case `subplotBuilder`, I'll pass a wrapper which clarifies what the intent of the object is. Passing an `EnsembleSet` can result in many things, while the only goal is to be able to create a display friendly name. Thereby wrapping a list of `Ensemble` objects or an `EnsembleSet` into wrapper/class named `DisplayNameGenerator` - clarifies the intent of the argument.

Personally I prefer this in opposed to a callback function, for readability. In my opinion a callback function is a nice tool when the users are entirely free to provide a function for general classes/components, or in API's, but not in a case where we are in full control of the utility classes - as here.

If callback logic is preferred, I can change it to that.

Reply:
My initial thought was that `subplotBuilder` was going to be used by multiple other modules that want to create/view sub plots. Setting a very specific class as an input takes away flexibility since only that specific class can be used for providing ensemble display names and the way the names are generated is predefined. Hence, if there is no flexibility, why even move the code out into an additional class if the display name generation is already predefined (who decides how the names are generated? - `subplotBuilder` as it defines its input types - so why not move the code there then? It already is a responsibility wrapper, naming is just another responsibility only used by itself, and according to SRP responsibilities should be grouped). 

On the other hand, if we create ensemble display names always in the same manner (only show case names in case of duplicates), the class implementation could become more valid to me as it might be a class used multiple places and a THING in our application (it's no longer any of `subplotBuilder`'s very own responsibilities). However, one could argue as well that the code might be moved into the `EnsembleSet` class then. The question could be boiled down then to whose responsibility it should be to provide a function for generating display names. Subplot builder, `EnsembleSet`, the consuming module (via a callback), or a responsibility wrapper around ensemble naming functionality - but we are not there yet.

As `subplotBuilder` is a very specific thing at the moment and very much specialized for `SimulationTimeSeries`, I don't see it being used elsewhere if its current implementation does not change. Hence, the flexibility argument loses its significance. Moreover, we haven't decided yet how and if we want to have application-wide advanced ensemble display naming implemented.

All in all, it comes down to the question if the `subplotBuilder` or the module should have the responsibility for generating display names, and in case of the latter if it should outsource some of its responsibility into another class, and if that class is useful enough to have the right to exist. I am struggling with this as the class is only having one single function (raising the question if the outsourcing is worth it, also when do we outsource/split responsibility in general and in what scope - something we can of course discuss in a group and find a common best practice). If you see a big advantage of having that extra class, I would be fine with it as long as this tight coupling between `subplotBuilder` and `SimulationTimeSeriesMatrix` exists (the responsibility lies in the module). As soon as the builder becomes more general/more widely used, we should revisit this issue as the responsibility question might come up again.